ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/QueryAnalyzer.java

```diff
@@ -97,177 +78,4 @@ public Analysis analyze(
 
     return analysis;
   }
-
-  public AggregateAnalysis analyzeAggregate(final Query query, final Analysis analysis) {
```

Comment:
Analysis of aggregates is now delayed until the creation of an `AggregateNode` in the `LogicalPlanner`.  This is necessary as this code needs any `*`s to have been resolved.

Reply:
I think this is just a continuation of a trend Tim started with UDTFs.  Trying to extract everything up front in the analyzer, and then using this to build the logical plan, is probably not the best approach.  It often involves duplicating logic between the analyser and plan, and if we just delay the analysis till we're building the plan we can take advantage of being able to ask parent nodes for things, i.e. we can build something more object oriented and take advantage of polymorphism, rather than `if` statements.

With regards to the resolution of `*` for GROUP BY:  It's valid to have `SELECT * FROM X GROUP BY ?,?,?` just so long as the GROUP BY contains all the columns in the source.  There's a QTT test covering this somewhere.  But we don't know what columns exist in the source until we've built the plan: the source could be a join for example. Hence we now delay this analysis till later.